apps/roam/scripts/compile.ts

```diff
@@ -132,6 +132,7 @@ export const compile = ({
       define: {
         "process.env.SUPABASE_URL": `"${process.env.SUPABASE_URL}"`,
         "process.env.SUPABASE_ANON_KEY": `"${process.env.SUPABASE_ANON_KEY}"`,
+        "process.env.NEXT_API_ROOT": `"${process.env.NEXT_API_ROOT || ""}"`,
```

Comment:
I understand you reference [this file](https://github.com/DiscourseGraphs/discourse-graph/blob/ENG-520-supabase-dev-instructions/packages/database/README.md), which says,

> If you are working on frontend code, you may use the environment variable NEXT_API_ROOT (forthcoming) to get the api of the active vercel branch instance (as given in the [vercel interface](https://vercel.com/discourse-graphs/discourse-graph/deployments).)

By "API" do you mean the "API URL"?

Is the indented workflow:
- publish a branch/create a PR
- log in to vercel/use the CLI/navigate to the PR
- find the branch's URL
- add that URL + "/api" to your `.env`/vercel `env` as the value for `NEXT_API_ROOT`?

Is there a plan to update all of the other places in the code that the API URL is referenced?

Reply:
Yes, API URL. Will rephrase.
Correct on workflow. That is in the case where you want to use your local dev code with the vercel branch (talking to the supabase branch.)
And yes, making a global update would be relevant. Presumably in a separate task.

Reply:
Introducing it now with no mention or reference to a future task to update the rest of the code will likely cause issues and confusion when someone tries to use it (as they would expect the API calls to use `NEXT_API_ROOT`when they don't, which wouldn't be fun to debug).  If you could go ahead and create the future task, assign it to yourself, and link it here, then we can merge this.

Reply:
> That is in the case where you want to use your local dev code with the vercel branch

It's also worth noting that this case is probably quite rare.  Usually people will use `localhost`.  Maybe useful when you need to redirect a third party to a live URL, but currently something like ngrok is a better dev experience because you don't have to wait for the vercel build every commit.

Reply:
Good, created the task https://linear.app/discourse-graphs/issue/ENG-585/generalize-usage-of-nextapiroot